### ISSUE TITLE
javers.findChanges() throws java.lang.IllegalArgumentException: Version is not a positive number.

### ISSUE THREAD

**Issue by ALP011216** [javers/javers#1121]
I want to integrate Javers on a database which is already in use.

If I mark an entity as deleted with `javers.commitShallowDelete` (for example Customer) for which no `INITIAL` snapshot is yet stored,
```
javers.findChanges(QueryBuilder.byClass(Customer.class).build());
```
throws the following exception:

```
java.lang.IllegalArgumentException: Version is not a positive number.
	at org.javers.common.validation.Validate.argumentCheck(Validate.java:43)
	at org.javers.repository.api.SnapshotIdentifier.<init>(SnapshotIdentifier.java:16)
	at org.javers.repository.api.SnapshotIdentifier.previous(SnapshotIdentifier.java:35)
	at org.javers.core.snapshot.SnapshotDiffer.calculateDiffs(SnapshotDiffer.java:43)
	at org.javers.repository.api.JaversExtendedRepository.getChangesIntroducedBySnapshots(JaversExtendedRepository.java:181)
	at org.javers.repository.api.JaversExtendedRepository.getChangeHistory(JaversExtendedRepository.java:49)
	at org.javers.repository.jql.ChangesQueryRunner.queryForChanges(ChangesQueryRunner.java:34)
	at org.javers.repository.jql.QueryRunner.queryForChanges(QueryRunner.java:46)
	at org.javers.core.JaversCore.findChanges(JaversCore.java:206)
```

**Javers' Version**
6.2.3

**Comment by anynamus** [comment]
I experienced the exact same issue with Javers 6.2.2

I would appreciate very much to know how to prevent this without having to create an initial version first.

**Comment by bartoszwalacik** [comment]
please follow https://github.com/javers/javers/blob/master/CONTRIBUTING.md#guidelines-for-bug-reporting

**Comment by anynamus** [comment]
Here is some code that reproduces the error:

https://github.com/anynamus/javers-demo-issue-1121

**Comment by anynamus** [comment]
After investigation, the issue comes from the method `org.javers.core.snapshot.SnapshotDiffer.calculateDiffs(List<CdoSnapshot>, Map<SnapshotIdentifier, CdoSnapshot>)` :
```java
if (snapshot.isTerminal()) {
	CdoSnapshot previousSnapshot = previousSnapshots.get(SnapshotIdentifier.from(snapshot).previous());
	addTerminalChanges(changes, snapshot, previousSnapshot);
}
```
It assumes that the parameter `previousSnapshots` has more than one entry for the terminal state when calling the `previous()` method. Which is not necessarily the case if Javers was not integrated at the beginning of the project.

However, the method `addTerminalChanges` does not make this assumption (`if (previousSnapshot != null...`):
```java
private void addTerminalChanges(List<Change> changes, CdoSnapshot terminalSnapshot, CdoSnapshot previousSnapshot) {
	changes.add(new ObjectRemoved(terminalSnapshot.getGlobalId(), empty(), of(terminalSnapshot.getCommitMetadata())));
	if (previousSnapshot != null && javersCoreConfiguration.isTerminalChanges()) {
		Diff terminalDiff = diffFactory.create(snapshotGraph(previousSnapshot), snapshotGraph(terminalSnapshot), commitMetadata(terminalSnapshot));
		changes.addAll(terminalDiff.getChanges());
	}
}
```

So, my best guess is that the fix should be within the processing of the terminal snapshot within the `calculateDiffs(...)` method or within the `SnapshotIdentifier.previous()` method which could return null if `getVersion() - 1 < 1`. But, actually, I am not sure if this fix would have side effects elsewhere in the _Javers_ modules.

In any case, the result should be that the `Changes.prettyPrint()`output should be like follow:
```text
Changes:
Commit 3.00 done by never mind at 14 Jul. 2021, 06:57:46 :
* object removed: ca.anynamus.javers.demo.issue1121.model.Person/2
  - 'firstName' value unset
  - 'id' value '1' unset
  - 'lastName' value unset
```

**Comment by bartoszwalacik** [comment]
thanks for reporting but this issue won't be analyzed if you don't provide a failing test case - see https://github.com/javers/javers/blob/master/CONTRIBUTING.md#guidelines-for-bug-reporting

**Comment by narsereg** [comment]
@bartoszwalacik Hi. I experience the same problem. So this test is failing now. 

https://github.com/narsereg/javers/commit/68d67183bf4a3e2db5a0d3a63217d643db613dc5

**Comment by bartoszwalacik** [comment]
fixed in 6.5.0
